indico/modules/events/registration/util.py

```diff
@@ -491,6 +491,26 @@ def get_published_registrations(event, is_participant):
     return query.all()
 
 
+def get_hidden_registrations(event, is_participant):
+    """Get the number of hidden registrations for an event.
+
+    :param event: the `Event` to get registrations for
+    :param is_participant: whether the user accessing the registrations is a participant of the event
+    :return: list of `Registration` objects
+    """
+    query = (Registration.query.with_parent(event)
+             .filter(~Registration.is_publishable(is_participant),
```

Comment:
I think adding these two criteria to the filter here will do the job:

```
Registration.is_active,
Registration.state.in_((RegistrationState.complete, RegistrationState.unpaid)),
```

Reply:
Since you need this check in multiple places, maybe add a `is_state_publishable` hybrid property containing these two checks; that way you can use `Registration.is_state_publishable` here and `reg.is_state_publishable` in the checks for conference events